README.md

```diff
@@ -13,17 +13,32 @@ Detailed user documentation is available by running `help` within fish, and also
 
 ## Building
 
-fish requires a C++11 compiler. It builds successfully with g++ 4.8 or later, or with clang 3.3 or later.
+### Dependencies
 
-fish can be built using autotools or Xcode. autoconf 2.60 or later, as well as automake 1.13 or later, are required to build from git versions. These are not required to build from released tarballs.
+Compiling fish requires:
 
-fish depends on a curses implementation, such as ncurses. The headers and libraries are required for building.
+* a C++11 compiler. It builds successfully with g++ 4.8 or later, or with clang 3.3 or later. fish can be built using autotools or Xcode. autoconf 2.60 or later, as well as automake 1.13 or later, are required to build from git versions, but these are not required to build from the officially released tarballs.
+* the headers and libraries of a curses implementation such as ncurses.
+* gettext, for translation support.
+* Doxygen 1.8.7 or newer, for building the documentation.
 
-fish requires PCRE2 due to the regular expression support contained in the `string` builtin. A copy is included with the source code, and will be used automatically if it does not already exist on your system.
+At runtime, fish requires:
 
-fish requires gettext for translation support.
+* a curses implementation, such as ncurses.
+* gettext, if you need the localizations.
+* PCRE2, due to the regular expression support contained in the `string` builtin. A copy is included with the source code, and will be used automatically if it does not already exist on your system.
+* a number of common UNIX utilities: coreutils (at least ls, seq, rm, mktemp), hostname, grep, awk, sed, and getopt.
+* bc, the "basic calculator" program.
 
-Building the documentation requires Doxygen 1.8.7 or newer.
+Optional functionality should degrade gracefully. If fish ever threw an error about missing dependencies for an optional feature, it would be a bug in fish and should be reported on our GitHub. Optional features and dependencies include:
+
+* fish's builtin `test` syntax, which requires `diff` and `expect`. You don't need these, though, if you don't want to use our tests.
```

Comment:
Ah, that makes sense. I think the original text was a little ambiguous, but I'll clarify it when I get a moment later today.